presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorTlsTest.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "arrow/flight/types.h"
+#include "arrow/testing/gtest_util.h"
+#include "folly/init/Init.h"
+#include "glog/logging.h"
+#include "gtest/gtest.h"
+#include "presto_cpp/main/connectors/arrow_flight/ArrowFlightConnector.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/FlightConnectorTestBase.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/FlightPlanBuilder.h"
+#include "presto_cpp/main/connectors/arrow_flight/tests/utils/Utils.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/config/Config.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+
+namespace facebook::presto::connector::arrow_flight::test {
+
+namespace {
+
+namespace velox = facebook::velox;
+namespace core = facebook::velox::core;
+namespace exec = facebook::velox::exec;
+namespace connector = facebook::velox::connector;
+namespace flight = arrow::flight;
+
+using namespace facebook::presto::connector::arrow_flight;
+using namespace facebook::presto::connector::arrow_flight::test;
+using exec::test::AssertQueryBuilder;
+using exec::test::OperatorTestBase;
+
+class FlightConnectorTlsTestBase : public FlightWithServerTestBase {
+ protected:
+  explicit FlightConnectorTlsTestBase(
+      std::shared_ptr<velox::config::ConfigBase> config)
+      : FlightWithServerTestBase(std::move(config)) {}
+
+  flight::Location getServerLocation() override {
+    AFC_ASSIGN_OR_RAISE(
+        auto loc, flight::Location::ForGrpcTls(BIND_HOST, LISTEN_PORT));
+    return loc;
+  }
+
+  void setFlightServerOptions(
+      flight::FlightServerOptions* serverOptions) override {
+    flight::CertKeyPair tlsCertificate{
+        .pem_cert = readFile("./data/tls_certs/server.crt"),
+        .pem_key = readFile("./data/tls_certs/server.key")};
+    serverOptions->tls_certificates.push_back(tlsCertificate);
+  }
+
+  void executeTest(
+      bool isPositiveTest = true,
+      const std::string& expectedError = "") {
+    updateTable(
+        "sample-data",
+        makeArrowTable(
+            {"id"},
+            {makeNumericArray<arrow::Int64Type>(
+                {1, 12, 2, std::numeric_limits<int64_t>::max()})}));
+
+    auto idVec = makeFlatVector<int64_t>(
+        {1, 12, 2, std::numeric_limits<int64_t>::max()});
```

Comment:
Abstract a variable for {1, 12, 2, std::numeric_limits<int64_t>::max()} so that it can be used in the input and result vector.